### DISCUSSION TITLE
Support PlantUML in-chat rendering

### DISCUSSION THREAD

**Discussion by ghost** [open-webui/open-webui#18605] [Ideas]
**Is your feature request related to a problem? Please describe.**
Not a problem but an extension of the functionality

**Describe the solution you'd like**
It would be really nice to have rendered PlantUML diagrams directly on the chat. Sometimes I feel like Mermaid diagrams don't reach the expected output quality. However when asking models to create PlantUML diagrams they are really good at it but to check it, we would have to render them in a PlantUML server. It would be a nice touch to have them directly rendered in the chat. 

**Describe alternatives you've considered**
As I already described above, I have tried using mermaid to replace what I would normally do in PlantUML. Maybe I am the problem but I do not like the charts so much as I do with PlantUML diagrams. 

**Additional context**
Here a comparative example. I know it is a matter of taste, but I still think PlantUML has more features:

**PlantUML Example:**
![dT712W8X483Xkqynno8wx3x1GkSew0acdKhGSHbdeILwzpnOL1JKUjN_KvmkAueECHZZMKaEQ0csajnW5Iu6obh2JiNp4P37JK-2wXFFvkPgZ0sOCwmIMo6bcZNWOnye4cj-enP_Kkqa2lTP1Sl9QWgX9dbqToUdIC_bwsxvY1ADQz9JSf-UMDa6wAp4xlizlqhj5AahGPdU3G00](https://github.com/user-attachments/assets/0c2785c2-305c-49f7-9374-5c23cf7c63b7)

**Mermaid Example:**
![Screenshot from 2024-09-28 17-14-35](https://github.com/user-attachments/assets/bee4cc73-65ab-4d75-8cd1-179a13990546)

PS: Please Ignore if the diagrams are not perfect or accurate, they are AI generated for the purpose of showing the difference.

**Comment by magnuslarsen** [comment]
I agree with this. A configurable for which PlantUML server to use would also be nice (so that you can use your own self-hosted PlantUML server)
